bundle/src/main/java/com/adobe/acs/commons/mcp/model/AvailableProcessDefinitions.java

```diff
@@ -0,0 +1,85 @@
+/*
+ * Copyright 2017 Adobe.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.adobe.acs.commons.mcp.model;
+
+import com.adobe.acs.commons.mcp.HiddenProcessDefinition;
+import com.adobe.acs.commons.mcp.form.FieldComponent;
+import com.adobe.acs.commons.mcp.form.FormField;
+import com.adobe.acs.commons.mcp.ProcessDefinition;
+import com.adobe.cq.sightly.WCMUsePojo;
+import java.lang.reflect.Field;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.lang3.reflect.FieldUtils;
+import org.apache.sling.api.scripting.SlingScriptHelper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * Produce a list of available process definitions that can be started by the
+ * user.
+ */
+public class AvailableProcessDefinitions extends WCMUsePojo {
```

Comment:
It is very, very weird.  But the goal was to handle the ugliness in java so that the component side could be expressed cleanly.  Had to pay the price somewhere.  Sightly doesn't give you a lot of other options.